File: Atten/parse.py
import argparse
import os


def parse_args(args=None):
    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
    parser.add_argument("-batch_size", type=int, default=128, help="batch size")
    parser.add_argument("-dimension", type=int, default=512, help="vector dimensionality")
    parser.add_argument("-dropout", type=float, default=0.2, help="dropout rate")
    parser.add_argument("-train_file", default="../data/t_given_s_train.txt")
    parser.add_argument("-dev_file", default="../data/t_given_s_dev.txt")
    parser.add_argument("-test_file", default="../data/t_given_s_test.txt")
    parser.add_argument("-init_weight", type=float, default=0.1)
    parser.add_argument("-alpha", default=1, type=float)
    parser.add_argument("-start_halve", default=6, type=int)
    parser.add_argument("-max_length", default=100, type=int)
    parser.add_argument("-vocab_source", default=25010, type=int)
    parser.add_argument("-vocab_target", default=25010, type=int)
    parser.add_argument("-thres", default=5, type=int, help="gradient clipping thres")
    parser.add_argument("-max_iter", default=8, type=int, help="max number of iteration")
    parser.add_argument("-source_max_length", default=50)
    parser.add_argument("-layers", default=2, type=int)
    parser.add_argument("-saveFolder", default="save")
    parser.add_argument("-reverse", action="store_true")
    parser.add_argument("-gpu_index", default=1, type=int, help="the index of GPU to use")
    parser.add_argument("-saveModel", action="store_false")
    parser.add_argument("-dictPath", default="../data/movie_25000")

    args = parser.parse_args(args)
    args.save_prefix = os.path.join(args.saveFolder, "model")
    args.save_params_file = os.path.join(args.saveFolder, "params")
    args.output_file = os.path.join(args.saveFolder, "log")
    os.mkdir(args.saveFolder)
    print(args)
    return args


if __name__ == '__main__':
    parse_args()

File: Atten/parse.lua
require 'logroll'

local logger = logroll.print_logger()

local function parse_args()
    local cmd = torch.CmdLine()

    cmd:option("-batch_size", 128, "batch size")
    cmd:option("-dimension", 512, "vector dimensionality")
    cmd:option("-dropout", 0.2, "dropout rate")
    cmd:option("-train_file", "../data/t_given_s_train.txt", "")
    cmd:option("-dev_file", "../data/t_given_s_dev.txt", "")
    cmd:option("-test_file", "../data/t_given_s_test.txt", "")
    cmd:option("-init_weight", 0.1, "weight used in random uniform initializer as [-init_weight, init_weight]")
    cmd:option("-alpha", 1, "initial learning rate")
    cmd:option("-start_halve", 6, "when to start halving the learning rate. Set to -1 to disable the halving")
    cmd:option("-max_length", 100, "");
    cmd:option("-vocab_source", 25010, "source vocabulary size")
    cmd:option("-vocab_target", 25010, "target vocabulary size")
    cmd:option("-thres", 5, "gradient clipping thres")
    cmd:option("-max_iter", 8, "max number of iteration")
    cmd:option("-source_max_length", 50, "max length of source sentences")
    cmd:option("-target_max_length", 50, "max length of target sentences")
    cmd:option("-layers", 2, "number of lstm layers")
    cmd:option("-saveFolder", "save", "the folder to save models and parameters")
    cmd:option("-reverse", false, "whether to reverse the sources")
    cmd:option("-gpu_index", 1, "the index of GPU to use")
    cmd:option("-saveModel", true, "whether to save the trained model")
    cmd:option("-dictPath", "../data/movie_25000", "dictionary file")

    logger.info('Parsing cmdline arguments...')
    local params = cmd:parse(arg)

    params.save_prefix = params.saveFolder .. "/model"
    params.save_params_file = params.saveFolder .. "/params"
    params.output_file = params.saveFolder .. "/log"

    if nil == paths.mkdir(params.saveFolder) then
        logger.warn(string.format('saveFolder %s already exist', params.saveFolder))
    end

    logger.info(params)

    return params;
end

return parse_args
